### PR TITLE
Capturing runtime exceptions from `Computer.threadPoolForRemoting.submit(Runnable)`

### PR DESCRIPTION
**Pull request by jglick**
Spent a while trying to track down an issue which turned out to be an unchecked exception thrown out of `JnlpConnectionStateListener.afterChannel` and swallowed by `FutureTask`, which `ExceptionCatchingThreadFactory` does _not_ address.

### Proposed changelog entries

- Developer: better error logging for unexpected problems in `Computer.threadPoolForRemoting`.

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7284"></a>

### CODE REVIEW

File: core/src/main/java/hudson/util/ExceptionCatchingThreadFactory.java
Comment (jglick) [reply core/src/main/java/hudson/util/ExceptionCatchingThreadFactory.java]: Oh, I normally do qualify Javadoc references, must have missed this one. Thanks.

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by NotMyFault** [comment]
@Vlatombe do you mind taking a look, please? I was holding off a merge, because Jesse requested your explicit review.

**Review by Vlatombe — APPROVED** [review]

**Pull request merged**
